tests/upgrades/conftest.py

```diff
@@ -260,11 +260,13 @@ def test_something_post_upgrade(pre_upgrade_data):
         start_index = test_node_id.find('[') + 1
         end_index = test_node_id.find(']')
         extracted_value = test_node_id[start_index:end_index]
-        upgrade_data[extracted_value] = _read_test_data(test_node_id)
+        if request.param in extracted_value:
+            upgrade_data[extracted_value] = _read_test_data(test_node_id)
     if len(upgrade_data) == 1:
         param_value = next(iter(upgrade_data.values()))
     else:
-        param_value = upgrade_data.get(request.param)
+        network_type = dict(request.keywords.node.user_properties).get('SatelliteNetworkType')
```

Comment:
The network type is not always `SatelliteNetworkType`, it is first checked in tests param and if not then only in SatelliteNetworkType

Reply:
The logic is that `extracted_value` can be anything like `rhel7`, `rhel7-ipv4`, `rhel7-ipv6` and other rhel variants.
Also `request.param` can be anything like that - with or without ipvX. When it is decorated _with_ ipvX, only that single value is added to the `upgrade_data` list due to the `if` on line 263. On the other hand, when `request.param` is not decorated with ipvX (i.e. just plain `rhel7`) the `upgrade_data` gets filled with everything that contains `rhel7`, like `['rhel7-ipv4', 'rhel7-ipv6']`. Only in that case we read the `SatelliteNetworkType` to resolve which one should be picked from `upgrade_data`.